### PR TITLE
Allow use of existing canvas.

### PR DESCRIPTION
**Pull request by ngoylufo**
Added some build restraints as well as a `.gitignore` file, cleaned up `html` and another example.

### CODE REVIEW

File: browser.go [hunk 1]
```diff
@@ -27,29 +28,42 @@ func Terminate() error {
 	return nil
 }
 
+func resolveCanvas() *dom.HTMLCanvasElement, bool {
+	canvas, ok := document.QuerySelector("#glfw").(*dom.HTMLCanvasElement)
+	if ok == false {
```
Comment (unitoftime) [inline browser.go]: nit (not important to fix here though): You can just do `if !ok {`

**Comment by unitoftime** [comment]
What are these files for? events.js, events.js.map

else looks fine. I added one nitpicky comment but don't worry about fixing that, more as a tip.

Thanks for the work!

File: browser.go (hunk 1 above)
Comment (ngoylufo) [reply browser.go]: I was feeling a bit more verbose I guess.

**Comment by ngoylufo** [comment]
> What are these files for? events.js, events.js.map

They were generated by gopherjs for browser tests/examples. The wasm scripts I added my self and they're needed for the examples to run.

The events can be ommited. As long as one has gopherjs installed they can regenerate those file.

File: browser.go (hunk 1 above)
Comment (unitoftime) [reply browser.go]: :)

**Comment by unitoftime** [comment]
Gotcha. Do you mind excluding those autogenerated files? If you don't want to I can probably figure out a way to do it. Other than that everything looks fine.

**Comment by unitoftime** [comment]
Thanks! Looks good!

**Pull request merged**